src/components/RegisterModal/index.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { FC } from 'react';
+import { Modal } from 'react-overlays';
+import { Text } from '@shared/Text';
+import { Button } from '@shared/Button';
+import { Input } from '@shared/Input';
+import { useFormik } from 'formik';
+import { RegisterModalPropsType } from '@/components/RegisterModal/type';
+
+export const RegisterModal: FC<RegisterModalPropsType> = ({
+  renderBackdrop,
+  loginOpen,
+  close,
+  isOpen,
+}) => {
+  const formik = useFormik({
+    initialValues: {
+      firstName: '',
+      secondName: '',
+      nickName: '',
+      location: '',
+      password: '',
+    },
+    onSubmit: ({ nickName, firstName, secondName, password, location }) => {
+      console.log(nickName, firstName, secondName, password, location);
```

Comment:
We'd want to update the application state here somehow to set the user as logged in.